doc/tutorials/contributing-clion.md

```diff
@@ -0,0 +1,183 @@
+  _Markdown to Shell Recorder_-tool).
+
+## Testing
+
+The most thorough way to test your changes is to run all tests. Therefore
+navigate to your run-configurations (Run -> Edit Configurations...) and look
+for the entry "run*all". There, "run_all" should be selected as \_Executable*,
+"kdb" as _Target_. Now you can execute this run-configuration which will run
+all enabled tests!
+
+If you want to test various _kdb_-methods separately, you can create your
+own run-configurations. Add a new one by clicking on the "+"-sign on the
+top left of the "Edit Configurations..."-window and name it. Here both
+_Executable_ and _Target_ should have "kdb" selected. If you for example
+want to test `kdb info dump`, write "info dump" next to _Program arguments_.
+That's it, now you can just test this part of _kdb_.
+
+For further information please read [this](/doc/TESTING.md).
+
+## Commiting your changes
+
+Once you are satisfied with your changes, you have to commit them to your
+forked repository. We'll use the terminal for working with git. By
+convention, such commits shouldn't be too large, otherwise it will become
+difficult to revert some small changes if they are not working as intended.
+To be able to commit code to your repository, you have to configure your
+local git-installation to use your GitHub-credentials. You can find
+information about how to do this
+[here](https://help.github.com/en/articles/set-up-git).
+
+After you've set up your git-configuration, you can continue with uploading
+your changes. By default, you are in the master-branch of your repository.
+First, you should never directly work on the master branch, since only
+working code is expected to be there. This means, we now create a branch in
+our repository where our code changes will be published into. To do this,
+open a terminal, navigate to the root directory of your local code and type:
+
+```sh
+git branch testbranch
+git checkout testbranch
+```
+
+Here we create a new branch called _testbranch_ and switch to it as our
+working branch (of course you can name it as you want). The next step is
+to add all files you've changed and also want in your commit to the _stage_.
+Suppose we've changed how `kdb get` works, we now have to add it to our files
+we want to commit (you can add several files for a commit too):
+
+```sh
+git add ./src/tools/kdb/get.cpp
+```
+
+Now we've staged our modified file for our commit. The final step is to
+actually commit it to your online repository. Therefore type:
+
+```sh
+git commit -m "Added functionallity to kdb get. Closes #1234"
+git push origin testbranch
+```
+
+With this, you've published your changes to your remote branch of your
+repository (Assuming you are fixing an issue with the number "1234", otherwise
+omit this text). The next step is merging this change into the original
+repository.
+
+## Creating a Pull Request
+
+This step is most easily done using a browser. Open the web page of the
```

Comment:
The built in Github support of Clion [covers creating PRs](https://www.jetbrains.com/help/clion/contribute-to-projects.html#create-pull-request), but I never tried it myself

Reply:
This can be added as extra information but it is good to also write that it is possible via web-interface.